### ISSUE TITLE
Allow me to disable SA16xx warnings for internal and private members

### ISSUE THREAD

**Issue by dotMorten** [DotNetAnalyzers/StyleCopAnalyzers#3352]
I realize there's a way to turn off requiring documenting internal members, but documenting internal members is useful so we get some intellisense for developers working directly with the code. Problem is if you turn on all the SA16xx analyzers, but you only provide a `<summary>` tag, you'll get warnings that it might not follow the proper pattern, that parameters aren't documented, punctuation might not be right, returns are missing etc.

A lot of the SA16xx analyzers are mostly great for generating good public API documentation, and very helpful for ensuring there's consistent documentation in the generated API doc, that really aren't so useful to be too nit-picky about for internal dev-to-dev, and if methods and parameters are self-explaining, requiring devs to fill this is in tedious work that serves no real purpose otherwise. Often times it really only serves a purpose for the generated documentation so empty fields or inconsistent wording and punctuation doesn't happen.

Of course in a perfect world internal and external doc would be nice, but it both becomes a resource issue, and with a large existing code base, it because prohibitively hard to turn on hard requirements for XML doc, just so we can have good public API doc, because it would require us to either delete or fix all internal documentation as well.

**Comment by sharwell** [comment]
Can you provide specific examples of code and the resulting diagnostics?

**Comment by dotMorten** [comment]
@sharwell It's pretty much any I can think of, except SA1615.
Example:
```cs
using System;

namespace ClassLib1
{
    /// <summary>
    /// Test class.
    /// </summary>
    public class Class1
    {
        /// <summary>
        /// Non-standard consttructor documentation violating SA1642
        /// </summary>
        internal Class1()
        {
        }

        /// <summary>
        /// No punctuation
        /// </summary>
        internal void SA1629()
        {
        }

        /// <summary>
        /// Summary here.
        /// </summary>
        /// <returns></returns>
        internal int SA1616(object parameter) => 0;

        /// <summary>
        /// Summary here.
        /// </summary>
        /// <param name="parameter"></param>
        internal int SA1614(object parameter) => 0;

        /// <summary>
        /// Summary here.
        /// </summary>
        internal int SA1615() => 0; // This one is ok. Only warning on missing return if public

        /// <summary>
        /// A property value.
        /// </summary>
        internal int SA1623 { get; set; }

        /// <summary>
        /// Gets or sets a property value.
        /// </summary>
        /// <value></value>
        internal int SA1610 { get; set; }
    }
}
```
stylecop.json:
```json
{
	  "$schema": "https://raw.githubusercontent.com/DotNetAnalyzers/StyleCopAnalyzers/master/StyleCop.Analyzers/StyleCop.Analyzers/Settings/stylecop.schema.json",
	  "settings": {
			"documentationRules": {
				"documentInternalElements": false,
				"documentExposedElements": true,
				"documentPrivateElements": false,
				"documentPrivateFields": false,
				"xmlHeader": false
			},
      "orderingRules": {
        "elementOrder": [
          "constant"
        ],
        "usingDirectivesPlacement": "outsideNamespace"
      },
      "layoutRules": {
        "newlineAtEndOfFile": "allow"
      },
	    "indentation": {
	      "indentationSize": 4,
	      "tabSize": 4,
	      "useTabs": false
	    },
	    "readabilityRules": {
	    }
	  }
}
```

**Comment by sharwell** [comment]
Interesting. It's not a case I really thought about before. It does seem like it would take a fair amount of work to ensure the behavior is consistent.

**Comment by dotMorten** [comment]
I'd be fine with just being able to set the scope in the global suppressions, and leave the current behavior by default
